src/controllers/v1/match2/index.ts

```diff
@@ -200,18 +253,28 @@ export class Match2Controller extends Controller {
   @Response<BadRequest>(400, 'Request was invalid')
   @SuccessResponse('201')
   public async acceptMatch2OnChain(@Path() match2Id: UUID): Promise<TransactionResponse> {
-    const [match2] = await this.db.getMatch2(match2Id)
-    if (!match2) throw new NotFound('match2')
+    const [maybeMatch2] = await this.db.getMatch2(match2Id)
+    validatePreOnChain(maybeMatch2, 'Match2', {})
+    const match2 = maybeMatch2 as Match2Row
 
     const state = match2.state
 
-    if (state === 'acceptedFinal') throw new BadRequest(`Already ${'acceptedFinal'}`)
+    if (state !== 'proposed' && state !== 'acceptedA' && state !== 'acceptedB')
+      throw new BadRequest(`state should not be ${state}`)
 
-    const [demandA]: DemandRow[] = await this.db.getDemand(match2.demandA)
-    validatePreOnChain(demandA, 'demand_a', 'DemandA')
+    const [maybeDemandA] = await this.db.getDemand(match2.demandA)
+    validatePreOnChain(maybeDemandA, 'DemandA', {
+      subtype: 'demand_a',
+      state: match2.replaces ? 'allocated' : 'created',
+    })
+    const demandA = maybeDemandA as DemandRow
 
-    const [demandB]: DemandRow[] = await this.db.getDemand(match2.demandB)
-    validatePreOnChain(demandB, 'demand_b', 'DemandB')
+    const [maybeDemandB] = await this.db.getDemand(match2.demandB)
+    validatePreOnChain(maybeDemandB, 'DemandB', {
+      subtype: 'demand_b',
+      state: 'created',
+    })
+    const demandB = maybeDemandB as DemandRow
```

Comment:
We maybe should also validate the state of the `replaces` match2 in the case of a rematch. It should be `acceptedFinal`

Reply:
wait.. accept is not mine ..how did it get here ?